common/rpc/rpc.go

```diff
@@ -235,3 +253,103 @@ func (d *RPCFactory) dial(hostName string, tlsClientConfig *tls.Config) *grpc.Cl
 func (d *RPCFactory) GetTLSConfigProvider() encryption.TLSConfigProvider {
 	return d.tlsFactory
 }
+
+// CreateLocalFrontendHTTPClient gets or creates a cached frontend client.
+func (d *RPCFactory) CreateLocalFrontendHTTPClient() (*common.FrontendHTTPClient, error) {
```

Comment:
Is there any concern about returning an error here? I notice our methods to create gRPC listener / connections just log a fatal message and return nil.

Reply:
No, we call this while setting up dependencies. If this errors the server won't start.
I think that's the right behavior.